scripts/release.py

```diff
@@ -0,0 +1,231 @@
+"""
+This script automates the release process for a Python package.
+
+It will:
+- Add a git tag for the given version.
+- Remove the previous dist folder.
+- Create a build.
+- Ask the user to verify the build.
+- Upload the build to PyPI.
+- Push all git tags to the remote.
+- Create a draft release on GitHub using the version notes in CHANGELOG.md.
+
+Prerequisites:
+    - This must be run from the root of the repository.
+    - The repo must have a clean git working tree.
+    - The user must have the GITHUB_TOKEN environment variable set to a valid GitHub personal access token.
```

Comment:
Can you add a bullet for PyPI credentials? During the script, users will need to paste in a token from their password manager?

Reply:
No problem. Done!